baseplate_celery/__init__.py

```diff
@@ -0,0 +1,130 @@
+
+    The keys useful to :py:func:`celery_from_config` should be prefixed, e.g.
+    ``celery.service_name``, ``celery.broker_url``, etc. The ``prefix`` argument
+    specifies the prefix used to filter keys.
+
+    Supported keys:
+
+    * ``service_name``          (required): the name of service.
+    * ``broker_url``            (required): a Redis URL like ``redis://localhost/0``.
+    * ``db.url``                (optional): Postgres configs for backend database.
+    * ``db.credentials_secret`` (optional): the key used to retrieve the database credentials from ``secrets``
+
+    """
+    assert prefix.endswith(".")
+    parser = config.SpecParser(
+        {
+            "service_name": config.String,
+            "broker_url": config.String,
+            "db": config.DictOf(config.String),
+        }
+    )
+    options = parser.parse(prefix[:-1], app_config)
+    return options
+
+
+def short_task_name(task_name: str) -> str:
+    return task_name.split(".")[-1]
+
+
+def emit_celery_metric(context, task_name, **tags):
+    tags = {"task_name": short_task_name(task_name), **tags}
+    context.metrics.counter("celery_task", tags=tags).increment()
+    context.metrics.flush()
+
+
+class BaseplateRequest(Request):
+    def on_timeout(self, soft, timeout):
+        super().on_timeout(soft, timeout)
+        emit_celery_metric(
+            self.task._context, self.task.name, success=False, timeout=True
+        )
+        self._context.sentry.capture_message(
+            "Task {self.task.name} timed out after {timeout} seconds"
+        )
+
+
+class BaseplateTask(Task):
+    Request = BaseplateRequest
+    acks_late = True
+    track_started = True
+
+    _context = None
+
+    def __call__(self, *args, **kwargs):
+        baseplate = self._app._baseplate
+        self._context = baseplate.make_context_object()
+        with baseplate.make_server_span(
+            self._context, f"celery.{short_task_name(self.name)}"
+        ):
+            return super().__call__(*args, **kwargs)
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        emit_celery_metric(self._context, self.name, success=False)
+        return super().on_failure(exc, task_id, args, kwargs, einfo)
+
+    def on_success(self, retval, task_id, args, kwargs):
+        emit_celery_metric(self._context, self.name, success=True)
+        return super().on_success(retval, task_id, args, kwargs)
+
+
+class BaseplateCelery(Celery):
+    def __init__(self, *args, **kwargs):
+        self._app_config = None
+        self._baseplate = None
+
+        super().__init__(*args, **kwargs)
+        self.Task = BaseplateTask
+
+    def set_baseplate(self, baseplate: Baseplate):
+        self._baseplate = baseplate
+        self._app_config = parse_celery_configs(baseplate._app_config)
+
+        secrets = secrets_store_from_config(baseplate._app_config, timeout=60)
+        session_factory = SQLAlchemySession(secrets).parse("", self._app_config.db)
```

Comment:
what's this for? i thought we were using redis as the broker?

Reply:
This configures the result backend.  Deep originally set this up.  None of our current tasks use the result backend. 

[In the docs](https://github.com/reddit/baseplate-celery.py/pull/1/files/b30e5ec9c640c40ccd3b48b8c707d0375e6ff086#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R35) we instruct them to set a db result backend url in the config, as well as defining the a `credentials_secret` but I don't see that these `celery.db.*` values we reference are actually read anywhere.  Seems like this was a plan that never came to fruition?